packages/sdk-codegen/src/specConverter.ts

```diff
@@ -190,5 +246,313 @@ export const fixConversion = (
+): IConversionResults => {
+  return fixConversionObjects(JSON.parse(openApiSpec), JSON.parse(swaggerSpec))
+}
+
+/**
+ * Convert a swagger structure ref to an OpenAPI structure ref
+ *
+ * TODO update this to the typescript that understands replaceAll
+ * @param ref string reference to convert
+ */
+export const swapRef = (ref: string) =>
+  ref.replace(/#\/definitions\//g, '#/components/schemas/')
+
+/**
+ * Get the name of the structure from the reference string
+ * @param ref to partse
+ */
+export const structName = (ref: string) => {
+  if (!ref) return undefined
+  const parts = ref.split('/')
+  return parts[parts.length - 1]
+}
+
+/**
+ * Moves type, format, items to schema
+ * Converts collectionFormat to style
+ * @param param to convert
+ */
+export const convertParam = (param: any) => {
+  const schema: any = { type: param.type }
+  const result = { ...param }
+  delete result.type
+  if (param.format) {
+    schema.format = param.format
+    delete result.format
+  }
+  if (param.collectionFormat) {
+    result.style = openApiStyle(param.collectionFormat)
+    delete result.collectionFormat
+  }
+  if (param.items) {
+    schema.items = param.items
+    result.explode = false
+    delete result.items
+  }
+  result.schema = schema
+  return result
+}
+
+/**
+ * Convert responses for the right response formats
+ * @param responses response to convert
+ * @param formats mime formats
+ */
+export const convertResponses = (
+  responses: ArgValues,
+  formats = defaultMimeFormats
+) => {
+  Object.entries(responses).forEach(([code, response]) => {
+    responses[code] = {
+      description: response.description,
+    }
+    if (response.schema) {
+      const content = {}
+      formats.produces.forEach(
+        (format) => (content[format] = { schema: response.schema })
+      )
+      responses[code].content = content
+    }
+  })
+  return responses
+}
+
+/**
+ * Convert an operation
+ * Assign schemas in params, create request bodies, update $refs
+ * @param op operation to convert
+ * @param formats mime format consumer and producer
+ */
+export const convertOp = (op: ArgValues, formats = defaultMimeFormats) => {
+  // If keys need to be shuffled for file comparison reasons, uncomment this function
+  const moveKey = (_key: string, value?: any) => {
+    return value
+    // if (!(key in op)) return
+    // if (!value) value = op[key]
+    // if (value) {
+    //   delete op[key]
+    //   op[key] = value
+    // }
+    // return value
+  }
+
+  let { produces, consumes } = formats
+  const body = {}
+  if (op.produces) {
+    produces = op.produces
+    delete op.produces
+  }
+  if (op.consumes) {
+    consumes = op.consumes
+    delete op.consumes
+  }
+  if (op.parameters) {
+    // Copy parameters before rebuilding
+    const params = { ...op.parameters }
+    op.parameters = []
+    Object.values(params).forEach((p: any) => {
+      if (p.in === 'body') {
+        // const struct = structName(p.schema?.$ref)
```

Comment:
why the commented out code here?

Reply:
because I'm going to put requestBodies processing back in soon now that I understand it.